unity-renderer/Assets/UIComponents/Scripts/Components/BaseComponentView.cs

```diff
@@ -150,7 +150,8 @@ public virtual void OnLoseFocus()
     public virtual void Dispose()
     {
         DataStore.i.screen.size.OnChange -= OnScreenSizeModified;
-        if (!isDestroyed)
+        
+        if (!isDestroyed && !Application.isEditor)
```

Comment:
Why are we avoiding destruction in `Editor`?

Reply:
it is mainly to remove Console error logs asking to use `DestroyImmidiate`. But I can use `#if UNITY_EDITOR` if you prefer. Here is the log
```
Destroy may not be called from edit mode! Use DestroyImmediate instead.
Destroying an object in edit mode destroys it permanently.
UnityEngine.Object:Destroy (UnityEngine.Object)
BaseComponentView:Dispose () (at Assets/UIComponents/Scripts/Components/BaseComponentView.cs:155)
```

Reply:
I made a refactoring for this, by hiding  #if UNITY_EDITOR inside `DestroyInternally` method. It was quite clean, but then a lot of tests cases were falling, so I reverted to this solution :(

Reply:
I think we are hiding an underlying issue. We are getting that `DestroyImmediate` log because that destruction is happening after PlayMode is stopped, that shouldn't happen.